File: 240116/블럭쌓는 명령2/block-stacking-commands2.java
import java.util.Scanner;
public class Main {
    public static void main(String[] args) {
        // 여기에 코드를 작성해주세요.
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        int m = sc.nextInt();
        int[] arr = new int[101];
        int max_val = 0 ;
        for (int i = 0 ; i<m; i++){
            int a = sc.nextInt();
            int b = sc.nextInt();

            for(int j = a; j<=b; j++){
                arr[i]+=1;
            }

            for(int z = 1 ; z<n; z++){
                if (arr[z]>max_val){
                    max_val = arr[z];
                }
            }
        }
        System.out.print(max_val);
    }
}

File: 240116/블럭쌓는 명령2/block-stacking-commands2.py
n , k =map(int,input().split())


max_val = 0 
for _ in range(k):
    arr = [0] *(n+1)
    a,b = map(int,input().split())

    for i in range(a,b+1):
        arr[i] = 1
    
    if sum(arr) > max_val:
        max_val = sum(arr)

print(max_val)